docs/CPS/CPS-XXXX/README.md

```diff
@@ -0,0 +1,129 @@
+- Using Mithril to validate data requires an agreed standard 
+
+> [!WARNING]
+> 
+> This fragility in the API creates friction around each protocol upgrade, with wallets needing to synchronise development effort, which often goes through multiple iterations over the course of months. 
+
+### Lacking a Standard
+- Wallet users, particularly light wallets, have little control over the data source, and subsequently must trust the product team's decisions.
+- Users cannot choose their own data provider, independent of the wallet software, which could be self-hosted or from an alternative vendor.   
+- This complicates or inhibits products offering a full node experience, with a free and open source service, available for users to install then choose their wallet of choice.
+
+### Rudimentary Rollback Handling
+- The Cardano blockchain has the potential to roll back blocks as part of resolving a chain fork event. 
+- Wallets electing to store data fetched from the provider locally must implement a strategy for handling the rollback scenario, which is specific to the application and therefore a responsibility of the client to handle.
+- An API querying data from a store is unaware of the rollback occurring, so the client must implement a heavy-handed approach by continually testing the tip meets expectations.
+- The result of this approach is a varying level of quality of outcomes in wallets, plus extra complexity and network demand. 
+
+> [!NOTE]
+> See [What is a rollback](https://plutus-apps.readthedocs.io/en/latest/plutus/explanations/rollback.html) and [CIP-9](https://cips.cardano.org/cip/CIP-9) for detail on `securityParam`.
+
+### Inefficient Address Discovery
+- Scanning a [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#account-discovery) keychain for used addresses has the potential to be quite an intensive operation and is non-trivial to implement in the wallet.
+- Most APIs support query by address only, missing the simplified alternative approach of using payment and stake key credentials. Funds intentionally or inadvertently sent to an address that would otherwise not be discovered under BIP-44 discovery standards would be found with this approach.
+- Funds under the control of unknown [Franken Address](https://www.essentialcardano.io/glossary/franken-address) go undetected, despite the wallet having part ownership.
+
+## Use cases
+
+### Lace Extension
+
+#### What is it doing?
+- Transaction data in responses are **transmitted as JSON**, via a **custom schema**
+- Data is sourced from the combination of [Cardano DB Sync], Ouroboros Local State Queries, and custom Ouroboros Chain-Sync projections into the PostgreSQL DB.
+- **Polling the tip for changes**. An HTTP GET request is made to a webserver every 5 seconds, checking if there's a new tip in the [Cardano DB Sync] database. If so, requests are then made to fetch transactions that are associated with the provided list of addresses after a certain point.
```

Comment:
I remember discussing why we must fetch the tip every 5 seconds, something related to NFTs? I suggest we note it in this document since it is the most frequently made call.

Reply:
The polling rate was selected to just optimise the UX, rather than a specific use case, since we have sending statuses to update etc. We could back this off to 10 or more seconds without too much impact